src/ds/d400/d400-options.cpp

```diff
@@ -90,33 +98,33 @@ namespace librealsense
                 toggle->set(1);
         }
 
-        command cmd_get(ds::AUTO_CALIB);
-        cmd_get.param1 = 5;
-        std::vector<uint8_t> ret = _hwm.send(cmd_get);
-        if (ret.empty())
-            throw invalid_value_exception("auto_exposure_limit_option::query result is empty!");
-
-        command cmd(ds::AUTO_CALIB);
-        cmd.param1 = 4;
-        cmd.param2 = static_cast<int>(value);
-        cmd.param3 = *(reinterpret_cast<uint32_t*>(ret.data() + 4));
-        _hwm.send(cmd);
-        _record_action(*this);
+        if( _new_opcode )
+        {
+            set_using_new_opcode( value );
+        }
+        else
+        {
+            set_using_old_opcode( value );
+        }
+        _record_action( *this );
+        
     }
 
     float auto_exposure_limit_option::query() const
     {
-        command cmd(ds::AUTO_CALIB);
-        cmd.param1 = 5;
-
-        auto res = _hwm.send(cmd);
-        if (res.empty())
-            throw invalid_value_exception("auto_exposure_limit_option::query result is empty!");
-
-        auto ret = static_cast<float>(*(reinterpret_cast<uint32_t*>(res.data())));
-        if (ret< get_range().min || ret > get_range().max)
+        float ret;
```

Comment:
Make a habit of initializing all declared variables.
`float ret = get_range().def;`
or
`float ret = _new_opcode ? query_using_new_opcode() : query_using_old_opcode();`